frontend/src/queries/Query/Query.tsx

```diff
@@ -24,7 +35,7 @@ export function Query(props: QueryProps): JSX.Element {
     if (isLegacyQuery(query)) {
         component = <LegacyInsightQuery query={query} />
     } else if (isDataTableNode(query)) {
-        component = <DataTable query={query} setQuery={setQuery} />
+        component = <DataTable query={query} setQuery={setQuery} custom={custom} />
```

Comment:
I renamed it to `context` and `context.columns` for now. I think it makes the purpose more clear.

However I'm not too happy with this. Such a feature basically breaks the ability to copy/paste queries from certain pages. E.g. open the web performance query in the query editor. 

Thus this "web performance waterfall button" should either be a part of the standard list of things we can ask for, or implemented as part of a `WebPerformanceTable` that somehow nests `DataTable`... or then some special "linkTemplate" column type that's part of the query.

I promise to try really hard not to implement "JSX in JSON" and call it "Jason X"

Reply:
It might be possible to develop a "good enough" templating engine for most custom columns (think excel functions), yet I fear we will still need some kind of escape hatch out of "json query land" into "local react app land", for special cases.

The biggest group needing this will be app developers. A lot of interfaces might be best described as "data tables plus a few custom react component columns". Think of the [feedback app](https://app.posthog.com/app/15453)'s sidebar component. This view might benefit from a button that makes it easy to reply directly within the table. Good luck doing that any other way than a custom React component.

Also, what if installing a posthog app adds new column types that anyone can use... 🤯